backend/routes/auth.js

```diff
@@ -52,5 +54,45 @@ router.post("/signup", async (req, res) => {
   res.status(201).json(newUser);
 });
 
+// login routes!
+
+const rateLimit = require("express-rate-limit");
+
+const loginLimiter = rateLimit({
+  windowMs: 15 * 60 * 1000, // 15 minutes
+  max: 100, // Limit each IP to 10 login attempts per windowMs
+  message: { error: "Too many failed login attempts. Try again in 10 minutes" },
+});
+
+router.post("/login", loginLimiter, async (req, res) => {
+  const { username, password } = req.body;
+  const user = await prisma.user.findUnique({
+    where: { username },
+  });
+  if (!user) {
+    return res.status(400).json({ error: "Invalid username or password." });
+  }
+
+  const isValidPassword = await argon2.verify(user.password, password);
+  if (!isValidPassword) {
+    return res.status(400).json({ error: "Invalid username or password." });
+  }
+  req.session.userId = user.id;
```

Comment:
@SantiHi i looked up session-based authentication and i think you're missing the logic to generate sessionID. then `sessionID` can be stored in the User table: https://www.authgear.com/post/session-vs-token-authentication

<img width="782" alt="image" src="https://github.com/user-attachments/assets/eba86998-0e5a-4af2-b503-a8c8ea6d54c6" />